### DISCUSSION TITLE
One click debugging of adapter in VS

### DISCUSSION THREAD

**Discussion by nohwnd** [nunit/nunit3-vs-adapter#1130] [Ideas]
In vstest we were struggling a lot with debugging because there is vstest.console and testhost, and you always had to manually attach. So I made this simple tool that attaches for you, and all recent versions of vstest pick it up based on environment variable. 

Would you development workflow benefit from such thing? 

I've made a demo for xunit here: https://github.com/xunit/visualstudio.xunit/issues/389#issue-1942277644

**Comment by OsirisTerje** [comment]
@nohwnd That sounds cool, absolutely open to that.

NUnit has some built-in possibility for debugging, you can just add a runsettings command -- NUnit.Debug and it will stop inside the adapter (see https://hermit.no/debugging-the-nunit3testadapter-take-2/) , and there are some options to NUnit itself, but having a facility like you've done for XUnit there sounds nice too, so just go ahead!
